impl/src/main/java/impl/QuestionResourceImpl.java

```diff
@@ -76,9 +78,9 @@ public QuestionResourceImpl(QuestionDao questionDao) {
     }
 
     @Override
-    public Observable<List<Question>> getQuestionsBySearchQuery(@NotNull String searchQuery) {
-        if (searchQuery == null || searchQuery.isEmpty()) {
-            return error(new WebException(HttpResponseStatus.BAD_REQUEST, FAILED_TO_SEARCH_FOR_QUESTIONS));
+    public Observable<List<Question>> getQuestionsBySearchQuery(String searchQuery) {
```

Comment:
I think RW will resolve Optional<String> but I think that isNullOrEmpty is sufficient.